other/guess_the_number_search.py

```diff
@@ -0,0 +1,151 @@
+"""
+get the random number guessed by the computer by passing the lower,higher
+and the number to guess
+
+this solution works on divide and getting the half of number of previous and
+current, this depends on the number is low or high
+
+if the number is more than last lower and less than to the number to guess then
+the number is assigned to it, and same but opposite for higher number
+
+suppose lower is 0, higher is 1000 and the number to guess is 355
+then:
+    num = int((lower+higher)/2)
+    for above statement the function already declared as the get_avg(a,b)
+
+        [1]
+    get_avg(0,1000)  : 500
+    answer(500) : high
+        Now this value is passed to the answer function and that returns the
+        passed number is lower than the guess number or higher than the guess
+        number and also for equality
+
+        [2]
+    get_avg(0,500) : 250
+    answer(250) : low
+
+        [3]
+    get_avg(250,500) : 375
+    answer(375) : high
+
+        [4]
+    get_avg(375,250) : 312
+    answer(312) : low
+
+        [5]
+    get_avg(312,375) : 343
+    answer(343) : low
+
+        [6]
+    get_avg(343,375) : 359
+    answer(359) : high
+
+        [7]
+    get_avg(343,359) : 351
+    answer(351) : low
+
+        [8]
+    get_avg(351,359) : 355
+    answer(355) : same
+
+The number is found : 355
+
+>>> guess_the_number(10, 1000, 17)
+started...
+guess the number : 17
+details : [505, 257, 133, 71, 40, 25, 17]
+
+"""
+
+
+def temp_input_value(min_val: int = 10, max_val: int = 1000, option: int = True) -> int:
+    """
+    Temporary input values for tests
+
+    >>> temp_input_value(option=True)
+    10
+
+    >>> temp_input_value(option=False)
+    1000
+    """
+    if option is True:
+        return min_val
+    return max_val
+
+
+def get_avg(number_1: int, number_2: int) -> int:
+    """
+    Return the mid-number(whole) of two integers a and b
+    >>> get_avg(10,15)
+    12
+    >>> get_avg(20,300)
+    160
+    >>> get_avg("abcd",300)
+    Traceback (most recent call last):
+        ...
+    TypeError: can only concatenate str (not "int") to str
+    """
+    return int((number_1 + number_2) / 2)
+
+
+def guess_the_number(lower: int, higher: int, to_guess: int) -> None:
+    """
+    The `guess_the_number` function that guess the number by some operations
+    and using inner functions
+
+    >>> guess_the_number(10, 1000, 17)
+    started...
+    guess the number : 17
+    details : [505, 257, 133, 71, 40, 25, 17]
+
+    >>> guess_the_number(10, 1000, "a")
+    Traceback (most recent call last):
+        ...
+    TypeError: '>' not supported between instances of 'int' and 'str'
```

Comment:
Please add tests for zero, negative numbers, and floating point numbers.

A ValueError must be raised if to_guess is less than lower or greater than upper or lower is greater than upper.